src/ck-core/charm++.h

```diff
@@ -1315,45 +1308,6 @@ int CkRegisterEp(const std::string& name, CkCallFnPtr call, int msgIdx, int char
   return CkRegisterEpTemplated(combined.c_str(), call, msgIdx, chareIdx, ck_ep_flags);
 }
 
-CkMarshallMsg *CkAllocateMarshallMsgNoninline(int size,const CkEntryOptions *opts);
-inline CkMarshallMsg *CkAllocateMarshallMsg(int size,const CkEntryOptions *opts=NULL)
-{
-	if (opts==NULL) {
-	  CkMarshallMsg *newMemory = new (size,0)CkMarshallMsg;
-	  setMemoryTypeMessage(UsrToEnv(newMemory));
-	  return newMemory;
-	}
-	else return CkAllocateMarshallMsgNoninline(size,opts);
-}
-
-
-
-
-
-
-
-template <typename T> 
-inline T *CkAllocateMarshallMsgT(int size,const CkEntryOptions *opts) 
-{ 
-  int priobits = 0; 
-  if (opts!=NULL) priobits = opts->getPriorityBits(); 
-  //Allocate the message 
-  T *m=new (size,priobits)T; 
-  //Copy the user's priority data into the message 
-  envelope *env=UsrToEnv(m); 
-  setMemoryTypeMessage(env); 
-  if (opts!=NULL) { 
-    CmiMemcpy(env->getPrioPtr(),opts->getPriorityPtr(),env->getPrioBytes()); 
-    //Set the message's queueing type 
-    env->setQueueing((unsigned char)opts->getQueueing()); 
-  } 
-  return m; 
-} 
-
-
-
-
-
```

Comment:
A little concerned about this change, did you check if all the codes with `CkAllocateMarshallMsg*` calls now include `ckmarshall.h` instead of (or in addition to) `charm++.h`?

Reply:
Since `ckmarshall.h` is included in `charm++.h`, it shouldn't be necessary to include it separately. This separation _should_ only be relevant for files included in `charm++.h`, such as `ckfutures.h`; everything related to marshal messages may not be accessible yet otherwise, and I encountered three different issues related to this prior to this separation.

Reply:
Ah I missed that, thanks